Source/AwsCommonRuntimeKit/io/EventLoopGroup.swift

```diff
@@ -6,22 +6,18 @@ import Foundation
 
 public final class EventLoopGroup {
     public var rawValue: UnsafeMutablePointer<aws_event_loop_group>
-
-    public let shutDownOptions: ShutDownCallbackOptions?
-
     public init(threadCount: UInt16 = 0,
                 allocator: Allocator = defaultAllocator,
-                shutDownOptions: ShutDownCallbackOptions? = nil) {
-        let ptr = shutDownOptions?.toShutDownCPointer()
-        self.shutDownOptions = shutDownOptions
-
-        self.rawValue = aws_event_loop_group_new_default(allocator.rawValue, threadCount, ptr)
+                shutDownOptions: ShutDownCallbackOptions? = nil) throws {
+        guard let rawValue = withOptionalUnsafePointer(shutDownOptions?.getCShutdownOptions(), { shutdownOptionsPointer in
+            aws_event_loop_group_new_default(allocator.rawValue, threadCount, shutdownOptionsPointer)
+        }) else {
+            throw CommonRunTimeError.crtError(.makeFromLastError())
+        }
```

Comment:
there's a memory leak here if `aws_event_loop_group_new_default()` fails

`shutDownOptions?.getCShutdownOptions()` resulted in +1 on the refcount of shutdownOptions

but that callback will never fire, since creation failed, so the refcount will never "balance"